compiler/modules/front.syntax/preparse.bal

```diff
@@ -89,8 +89,16 @@ function preparseArrayTypeDesc(Tokenizer tok) returns boolean|err:Syntax {
         check tok.advance();
         _ = check tok.expectIdentifier();
     }
-    check tok.expect("[");
-
-    // SUBSET fixed length array types / tuples are not supported, is a td only if no token between `[` `]`
-    return tok.current() == "]";
+    while tok.current() == "[" {
```

Comment:
This doesn't look like it's on the right track. I would expect something more like preparseBracketed.